### PR TITLE
[SPARK-30437][SQL] Uneven spaces for some fields in EXPLAIN FORMATTED

### PR DESCRIPTION
**Pull request by amanomer**
### What changes were proposed in this pull request?
This PR is to match the output of `verboseStringWithOperatorId()` for Project, FIlter and Exchange node with other nodes.

### Why are the changes needed?
There are extra spaces at the end of some fields which is not similar to the output of other nodes.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Test cases updated.

### CODE REVIEW

**Comment by amanomer** [comment]
cc @srowen

**Comment by srowen** [comment]
It's there to make it line up for easier visual comparison. I'd leave that.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala [hunk 1]
```diff
@@ -49,9 +49,9 @@ trait HashJoin {
 
     s"""
        |(${ExplainUtils.getOpId(this)}) $nodeName ${ExplainUtils.getCodegenId(this)}
-       |Left keys: ${leftKeys}
-       |Right keys: ${rightKeys}
-       |Join condition: ${joinCondStr}
+       |Left keys      : ${leftKeys}
+       |Right keys     : ${rightKeys}
+       |Join condition : ${joinCondStr}
```
Comment (amanomer) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala]: @srowen Here we need this line up for better visualization?

Comment (srowen) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala]: I don't think the same logic applies, no. I don't think a change is necessary.

Comment (amanomer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala]: Got it. Thanks @srowen

**Pull request closed without merging**